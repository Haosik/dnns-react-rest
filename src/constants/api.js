export const API_URL = 'https://jsonplaceholder.typicode.com/';
export const PHOTOS_URL = `${API_URL}photos`;
