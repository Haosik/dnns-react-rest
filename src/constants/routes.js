export const ALBUM_ROUTE = '/album/';
