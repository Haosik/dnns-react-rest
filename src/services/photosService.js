import api from '../services/apiService';

const TOTAL_IMAGES = 300;

export const sortByAlbum = photos => {
  const slicedPhotos = photos.slice(0, TOTAL_IMAGES);
  const albumsHashMap = slicedPhotos.reduce((acc, el) => {
    const { albumId } = el;
    acc.hasOwnProperty(albumId) ? acc[albumId].push(el) : (acc[albumId] = [el]);
    return acc;
  }, {});
  return albumsHashMap;
};

// id argument for testing purposes (in other case 5000 items may fail test)
export const getPhotosFromApi = async id => {
  const resp = await api.getPhotos(id);
  const respPhotos = await resp.json();
  return respPhotos;
};

export default {
  sortByAlbum,
  getPhotosFromApi
};
