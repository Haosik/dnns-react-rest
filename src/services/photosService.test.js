import { sortByAlbum, getPhotosFromApi } from './photosService';
import { TOTAL_IMAGES } from './../constants/api';

describe('getPhotosFromApi', () => {
  it('should make API call and return data if api responded', async () => {
    expect(await getPhotosFromApi()).toMatchSnapshot();
  });
  it('should return limited number of items from API', async () => {
    expect(await getPhotosFromApi()).toHaveLength(TOTAL_IMAGES);
  });
});

describe('sortByAlbum', () => {
  const photos = [
    {
      albumId: 1,
      id: 1,
      thumbnailUrl: 'https://via.placeholder.com/150/92c952',
      title: 'accusamus beatae ad facilis cum similique qui sunt',
      url: 'https://via.placeholder.com/600/92c952'
    },
    {
      albumId: 99,
      id: 88,
      thumbnailUrl: 'https://via.placeholder.com/150/771796',
      title: 'reprehenderit est deserunt velit ipsam',
      url: 'https://via.placeholder.com/600/771796'
    },
    {
      albumId: 99,
      id: 199,
      thumbnailUrl: 'https://via.placeholder.com/150/771796',
      title: 'reprehenderit est deserunt velit ipsam',
      url: 'https://via.placeholder.com/600/771796'
    }
  ];

  const expected = {
    1: [
      {
        albumId: 1,
        id: 1,
        thumbnailUrl: 'https://via.placeholder.com/150/92c952',
        title: 'accusamus beatae ad facilis cum similique qui sunt',
        url: 'https://via.placeholder.com/600/92c952'
      }
    ],
    99: [
      {
        albumId: 99,
        id: 88,
        thumbnailUrl: 'https://via.placeholder.com/150/771796',
        title: 'reprehenderit est deserunt velit ipsam',
        url: 'https://via.placeholder.com/600/771796'
      },
      {
        albumId: 99,
        id: 199,
        thumbnailUrl: 'https://via.placeholder.com/150/771796',
        title: 'reprehenderit est deserunt velit ipsam',
        url: 'https://via.placeholder.com/600/771796'
      }
    ]
  };

  it('should return sorted by albumId object ', () => {
    expect(sortByAlbum(photos)).toEqual(expected);
  });
});
