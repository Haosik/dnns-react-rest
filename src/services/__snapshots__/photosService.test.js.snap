// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getPhotosFromApi should make api call and return data if api responded 1`] = `
Object {
  "albumId": 1,
  "id": 1,
  "thumbnailUrl": "https://via.placeholder.com/150/92c952",
  "title": "accusamus beatae ad facilis cum similique qui sunt",
  "url": "https://via.placeholder.com/600/92c952",
}
`;
